### DISCUSSION TITLE
Workflow inheritance - reusing workflow and task logic

### DISCUSSION THREAD

**Discussion by paweljasinski** [codingjoe/joeflow#35] [General]
We have 3 workflows with ~90% of the flow logic and tasks identical.
Initially we tried to reuse the common parts using inheritance.
This didn't work because of the way name and type attribute of a tasks are assigned.
We ended up repeating all the tasks in sub classes and calling explicitly using super().
Is there a better way?

**Comment by paweljasinski** [comment]
I think I have figured out the problem. The super workflow class has no edges. Without edges all the functions are not considered tasks by meta machinery and as a result are missing the name, type and workflow_cls attributes

**Reply by codingjoe** [reply]
Hi @paweljasinski, thanks for reaching out. Seems you have solved your own question. Should you have any other questions, please don't hesitate to reach out. Best Joe
